galsim/config/process.py

```diff
@@ -221,14 +246,17 @@ def Process(config, logger=None, njobs=1, job=1, new_params=None, except_abort=F
     # First thing to do is deep copy the input config to make sure we don't modify the original.
     config = CopyConfig(config)
 
+    # Import any modules if requested
+    ImportModules(config)
+
     # Process any template specifications in the dict.
     ProcessAllTemplates(config, logger)
 
     # Update using any new_params that are given:
     if new_params is not None:
         UpdateConfig(config, new_params)
 
-    # Import any modules if requested
+    # Do this again in case any new modules were added by the templates or command line params.
     ImportModules(config)
```

Comment:
I think I have seen cases where, depending on what people are doing, importing modules twice can cause errors.  Could that be a worry here?

Reply:
It shouldn't for any well behaved package.  I'd be very surprised if this caused problems for anyone.  Python normally allows a repeated import statement to basically act as a no op.

The reason for this is that we want both:

1. Named templates in one of the packages in the `modules` list to be available for `template` specifications.  (This is the use case for the imsim user config for instance)
2. Template files that include a `modules` directive should be loaded, so any registered things there are processed properly.  (We have example configs in the great3 and des subdirs that rely on this.)

So importing the modules needs to happen both before and after the template processing.